StoreMessageContentActivity/__tests__/handler.test.ts

```diff
@@ -0,0 +1,256 @@
+    );
+
+    expect(result.kind).toBe("SUCCESS");
+    if (result.kind === "SUCCESS") {
+      expect(result.blockedInboxOrChannels).toEqual([]);
+      expect(result.profile).toEqual(aRetrievedProfileWithAValidTimestamp);
+    }
+  });
+
+  it("should fail if activity input cannot be decoded", async () => {
+    const storeMessageContentActivityHandler = getStoreMessageContentActivityHandler(
+      profileModelMock as any,
+      messageModelMock as any,
+      {} as any,
+      aPastEmailModeSwitchLimitDate
+    );
+
+    const result = await storeMessageContentActivityHandler(
+      mockContext,
+      {} as any
+    );
+
+    expect(result.kind).toBe("FAILURE");
+    if (result.kind === "FAILURE") {
+      expect(result.reason).toEqual("BAD_DATA");
+    }
+  });
+
+  it("should throw an Error if there is an error while fetching profile", async () => {
+    findLastVersionByModelIdMock.mockImplementationOnce(() =>
+      fromLeft("Profile fetch error")
+    );
+    const storeMessageContentActivityHandler = getStoreMessageContentActivityHandler(
+      profileModelMock as any,
+      messageModelMock as any,
+      {} as any,
+      aPastEmailModeSwitchLimitDate
+    );
+
+    await expect(
+      storeMessageContentActivityHandler(mockContext, aCreatedMessageEvent)
+    ).rejects.toThrow();
+  });
+
+  it("should return a failure if no profile was found", async () => {
+    findLastVersionByModelIdMock.mockImplementationOnce(() =>
+      taskEither.of(none)
+    );
+    const storeMessageContentActivityHandler = getStoreMessageContentActivityHandler(
+      profileModelMock as any,
+      messageModelMock as any,
+      {} as any,
+      aPastEmailModeSwitchLimitDate
+    );
+
+    const result = await storeMessageContentActivityHandler(
+      mockContext,
+      aCreatedMessageEvent
+    );
+
+    expect(result.kind).toBe("FAILURE");
+    if (result.kind === "FAILURE") {
+      expect(result.reason).toEqual("PROFILE_NOT_FOUND");
+    }
+  });
+
+  it("should return a failure if inbox is not enabled", async () => {
+    findLastVersionByModelIdMock.mockImplementationOnce(() =>
+      taskEither.of(some({ ...aRetrievedProfile, isInboxEnabled: false }))
+    );
+    const storeMessageContentActivityHandler = getStoreMessageContentActivityHandler(
+      profileModelMock as any,
+      messageModelMock as any,
+      {} as any,
+      aPastEmailModeSwitchLimitDate
+    );
+
+    const result = await storeMessageContentActivityHandler(
+      mockContext,
+      aCreatedMessageEvent
+    );
+
+    expect(result.kind).toBe("FAILURE");
+    if (result.kind === "FAILURE") {
+      expect(result.reason).toEqual("MASTER_INBOX_DISABLED");
+    }
+  });
+
+  it("should return a failure if message sender is blocked", async () => {
+    findLastVersionByModelIdMock.mockImplementationOnce(() =>
+      taskEither.of(
+        some({
+          ...aRetrievedProfile,
+          blockedInboxOrChannels: { agid: BlockedInboxOrChannelEnum.INBOX }
```

Comment:
use anonymous service ID, and i think that the value should be an array
```suggestion
          blockedInboxOrChannels: { myServiceId: [BlockedInboxOrChannelEnum.INBOX] }
```